com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/liquibase/PopulateGUIDColumnConfigurationRegistryChange.java

```diff
@@ -0,0 +1,125 @@
+
+        HashMap<Long, String> result = new HashMap<>();
+        CloudControllerClient cfClient = getCFClient();
+        List<CloudSpace> spaces = getSpaces(cfClient);
+
+        String org, space;
+        for (Long id : retrievedData.keySet()) {
+            org = retrievedData.get(id).getOrg();
+            space = retrievedData.get(id).getSpace();
+            if (StringUtils.isEmpty(space)) {
+                continue;
+            }
+
+            if (StringUtils.isEmpty(org)) {
+                result.put(id, space);
+            } else {
+                String spaceGuid = getSpaceGuid(org, space, spaces);
+                result.put(id, spaceGuid);
+            }
+            
+        }
+        return result;
+    }
+
+    protected List<CloudSpace> getSpaces(CloudControllerClient cfClient) {
+        return cfClient.getSpaces();
+    }
+
+    protected String getSpaceGuid(String org, String space, List<CloudSpace> spaces) {
+
+        for (CloudSpace cloudSpace : spaces) {
+            if (org.equals(cloudSpace.getOrganization().getName()) && space.equals(cloudSpace.getName())) {
+                return cloudSpace.getMeta().getGuid().toString();
+            }
+        }
+
+        return null;
+    }
+
+    @Override
+    public Map<Long, CloudTarget> extractData(ResultSet resultSet) throws SQLException {
+
+        Map<Long, CloudTarget> result = new HashMap<>();
+        CloudTarget cloudTarget;
+        while (resultSet.next()) {
+            long id = resultSet.getLong("ID");
+            String targetSpace = resultSet.getString("TARGET_SPACE");
+            String targetOrg = resultSet.getString("TARGET_ORG");
+            cloudTarget = new CloudTarget(targetOrg, targetSpace);
+            result.put(id, cloudTarget);
+            logger.debug(
+                String.format("Retrieve data from row ID: '%s', TARGET_ORG: '%s' and TARGET_SPACE: '%s'", id, targetOrg, targetSpace));
+        }
+
+        return result;
+    }
+
+    @Override
+    public void setUpdateStatementParameters(PreparedStatement preparedStatement, Map<Long, String> transformedData) throws SQLException {
+
+        for (Map.Entry<Long, String> entry : transformedData.entrySet()) {
+            String space_id = entry.getValue();
+            preparedStatement.setString(1, space_id);
+            preparedStatement.setLong(2, entry.getKey());
+            preparedStatement.addBatch();
+            logger.debug(String.format("Executed update for row ID: '%s' , SPACE_ID: '%s' ", entry.getKey(), space_id));
+        }
+    }
+
+    @Override
+    public String getSelectStatement() {
+        return SELECT_STATEMENT;
+    }
+
+    @Override
+    public String getUpdateStatement() {
+        return UPDATE_STATEMENT;
+    }
+
+    @Override
+    public String getConfirmationMessage() {
+        return Messages.POPULATE_SPACE_ID_COLUMN;
+    }
+
+    protected CloudControllerClient getCFClient() {

```

Comment:
Why don't you use com.sap.cloud.lm.sl.cf.core.cf.CloudFoundryClientFactory here?

Reply:
In liquibase custom change dependency injection is now allowed but in this factory there are DP. Also with this client we cannot create XS client because we do not have visibility in CF project.